homework-selects/sql_questions.sql

```diff
@@ -1,57 +1,579 @@
+    COUNT(E.employee_id) AS "COUNT EMP"
+FROM departments D
+    INNER JOIN employees E
+        ON D.department_id = E.department_id
+    GROUP BY D.department_name
+	
 --11. the name of the department, average salary and number of employees working in that department who got commission.
+
+SELECT D.department_name, 
+    ROUND(AVG(E.salary), 2) AS "AVG SALARY",
+    COUNT(E.employee_id) AS "COUNT OF EMP WITH COMM"
+FROM departments D
+    INNER JOIN employees E
+        ON D.department_id = E.department_id
+    WHERE E.commission_pct is not null
+    GROUP BY D.department_name;
+	
 --12. job title and average salary of employees.
+
+SELECT J.job_title, AVG(E.salary) AS "AVG SALARY"
+FROM jobs J
+    INNER JOIN employees E USING (job_id)
+    GROUP BY J.job_title
+	
 --13. the country name, city, and number of those departments where at least 2 employees are working.
+
+SELECT country_name, city, COUNT(department_id)
+FROM locations
+    JOIN countries USING (country_id)
+    JOIN departments USING (location_id)
+    WHERE department_id IN (
+        SELECT department_id 
+            FROM employees
+            GROUP BY department_id
+                HAVING COUNT(department_id) > 1
+    )
+    GROUP BY country_name, city
+	
 --14. the employee ID, job name, number of days worked in for all those jobs in department 80.
+
+SELECT employee_id, 
+    job_title, 
+    end_date - start_date DAYS FROM job_history 
+    JOIN jobs USING (job_id)
+WHERE department_id = 80;
+
 --15. the name ( first name and last name ) for those employees who gets more salary than the employee whose ID is 163.
+
+SELECT first_name || ' ' || last_name as "FULL NAME"
+FROM employees 
+    WHERE salary > (
+        SELECT salary
+        FROM employees
+            WHERE employee_id = 163
+        );
+
 --16. the employee id, employee name (first name and last name ) for all employees who earn more than the average salary.
+
+SELECT employee_id, 
+	first_name || ' ' || last_name as "FULL NAME" 
+FROM employees 
+    WHERE salary > (
+        SELECT avg(salary)
+        FROM employees
+    );
+	
 --17. the employee name ( first name and last name ), employee id and salary of all employees who report to Payam.
+
+SELECT first_name || ' ' || last_name as "FULL NAME", 
+	employee_id, 
+	salary
+FROM employees 
+    WHERE manager_id IN (
+        SELECT employee_id
+        FROM employees
+        WHERE first_name = 'Payam'
+    );
+	
 --18. the department number, name ( first name and last name ), job and department name for all employees in the Finance department.
+
+SELECT 
+	department_id,
+	first_name || ' ' || last_name as "FULL NAME", 
+	job_title, 
+	department_name 
+FROM employees
+    JOIN departments USING (department_id)
+    JOIN jobs USING (job_id)
+    WHERE department_name = 'Finance'
+	
 --19. all the information of an employee whose id is any of the number 134, 159 and 183.
+
+SELECT * 
+FROM employees 
+    WHERE employee_id IN (134, 159, 183);
```

Comment:
we need to print information only of one person

Reply:
Good, fixed by using FETCH